mptcpd tests fail to compile with clang

**Describe the bug**
Some mptcpd tests fail to compile with the [clang](http://clang.llvm.org/) C compiler.
* `test-plugin` fails with _undefined reference_ errors:
```make
  CCLD     test-plugin
test_plugin-test-plugin.o: In function `test_plugin_dispatch':
/home/ossama/MPTCP/OTC/mptcpd/tests/test-plugin.c:131: undefined reference to `call_plugin_ops'
/home/ossama/MPTCP/OTC/mptcpd/tests/test-plugin.c:141: undefined reference to `call_plugin_ops'
/home/ossama/MPTCP/OTC/mptcpd/tests/test-plugin.c:151: undefined reference to `call_plugin_ops'
test_plugin-test-plugin.o: In function `run_plugin_load':
/home/ossama/MPTCP/OTC/mptcpd/tests/test-plugin.c:40: undefined reference to `call_plugin_ops'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
* `test-path-manager` fails with a _missing field initializer_ error
```make
  CC       test_path_manager-test-path-manager.o
test-path-manager.c:119:40: error: missing field 'pm' initializer
      [-Werror,-Wmissing-field-initializers]
        struct test_info info = { NULL };
                                       ^
1 error generated.
```

**To Reproduce**
Steps to reproduce the behavior:
1. ./configure CC=clang CXX=clang++
2. make check
3. `tests/test-plugin` will fail to link.
4. Apply the following patch :
```diff
diff --git a/tests/test-plugin.h b/tests/test-plugin.h
index d53634a..61584ed 100644
--- a/tests/test-plugin.h
+++ b/tests/test-plugin.h
@@ -275,7 +275,7 @@ inline bool mptcpd_addr_is_equal(struct mptcpd_addr const *lhs,
         return false;
 }
 
-inline void call_plugin_ops(struct plugin_call_count const *count,
+extern void call_plugin_ops(struct plugin_call_count const *count,
                             char const *name,
                             mptcpd_cid_t connection_id,
                             mptcpd_aid_t laddr_id,
```
5. `tests/test-path-manager` will fail to compile.

**Expected behavior**
The entire mptcpd source distribution should build without error when using other C and C++ compilers, such as `clang`.

**Desktop (please complete the following information):**
 - Compiler: *clang 6.0.0*

--- a/tests/test-path-manager.c
+++ b/tests/test-path-manager.c
@@ -116,7 +116,7 @@ int main(void)
 
         static int argc = L_ARRAY_SIZE(argv);
 
-        struct test_info info = { NULL };
+        struct test_info info = { .config = NULL };
 
         info.config = mptcpd_config_create(argc, argv);
         assert(info.config);

--- a/tests/test-plugin.h
+++ b/tests/test-plugin.h
@@ -251,7 +251,7 @@ struct mptcpd_addr const test_raddr_4 = {
 /**
  * @brief Compare equality of two @c mptcpd_addr objects.
  */
-inline bool mptcpd_addr_is_equal(struct mptcpd_addr const *lhs,
+extern bool mptcpd_addr_is_equal(struct mptcpd_addr const *lhs,
                                  struct mptcpd_addr const *rhs)
 {
         if (lhs->address.family == rhs->address.family
@@ -275,7 +275,7 @@ inline bool mptcpd_addr_is_equal(struct mptcpd_addr const *lhs,
         return false;
 }
 
-inline void call_plugin_ops(struct plugin_call_count const *count,
+extern void call_plugin_ops(struct plugin_call_count const *count,
                             char const *name,
                             mptcpd_cid_t connection_id,
                             mptcpd_aid_t laddr_id,

